### PR TITLE
[hpt-842] Create CONTENTdm wrapper for preingest task to create YML file.

### PR DESCRIPTION
**Pull request by andjsmit**
Also makes slight change to ingest code so URLs can be used for file uploading.

### CODE REVIEW

**Review by aploshay — APPROVED** [review]
Looks good to me!  Suggest squash merge to collapse commits.

File: app/models/contentdm_export.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    # Fix file paths for IUPUI exports
+    #
+    # @param [String] path given from IUPUI contentDM export
+    # @return [String] corrected path using new API port for IUPUI contentDM
+    def fix_path_iupui(path)
+      # IUPUI CDM no longer provides API on port 445
+      # The API is now available on port 2012
+      # Also needs to replace &amp; with just &
+      path = path.sub(/445\/cgi-bin/, '2012/cgi-bin')
+      path.sub('&amp;', '&')
```
Comment (mwoodiupui) [inline app/models/contentdm_export.rb]: Should this perhaps use CGI.unescapeHTML()? or Nokogiri::HTML.parse()?

File: app/models/contentdm_export.rb [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        array << c
+      end
+      array
+    end
+
+    # Fix file paths for IUPUI exports
+    #
+    # @param [String] path given from IUPUI contentDM export
+    # @return [String] corrected path using new API port for IUPUI contentDM
+    def fix_path_iupui(path)
```
Comment (mwoodiupui) [inline app/models/contentdm_export.rb:115]: I'm uneasy about the existence of this method.  It's not at all portable to use this the way we are.  If this sort of mishegoss is common at CDM sites, maybe this class needs a configurable filter, with this method factored out to a separate class.

Comment (andjsmit) [reply app/models/contentdm_export.rb:115]: This is definitely not common to CDM sites. In our case, when the system was moved between servers, the port changed. Unfortunately, the metadata retrieved from the export API does not reflect the change.

File: app/models/contentdm_export.rb (hunk 1 above)
Comment (andjsmit) [reply app/models/contentdm_export.rb]: Updated code to use CGI.unescape.

File: app/models/contentdm_export.rb (hunk 2 above)
Comment (mwoodiupui) [reply app/models/contentdm_export.rb:115]: OK, then, CDM is buggy (recording a value that should be synthesized) and we have to deal with it.  At some point, to make the code portable, we should factor this fix out.  But it is not of immediate concern.

**Review by mwoodiupui — APPROVED** [review]
+1 by inspection

**Pull request merged**